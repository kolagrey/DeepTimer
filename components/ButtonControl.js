import React from 'react';
import { Button } from 'react-native-paper';
import PropTypes from 'prop-types';
import styles from '../styles';

const ButtonControl = (props) => {
    const { disabled, speed, controlSpeed, buttonAction, buttonText } = props;
    return (
        <Button disabled={disabled} style={speed === controlSpeed ? styles.activeSpeedButton : styles.inActiveSpeedButton} contentStyle={styles.speedButtonContentStyle} mode="outlined" onPress={() => buttonAction({ speed: controlSpeed })}>
            {buttonText}
        </Button>
    );
};

ButtonControl.propTypes = {
    disabled: PropTypes.bool,
    speed: PropTypes.number,
    controlSpeed: PropTypes.number,
    buttonAction: PropTypes.func,
    buttonText: PropTypes.string
};

export default ButtonControl;