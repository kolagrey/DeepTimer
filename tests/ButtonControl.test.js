import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import ButtonControl from '../components/ButtonControl';


describe('<ButtonControl/>', () => {

    test('renders correctly', () => {
        const tree = renderer.create(<ButtonControl/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("is of type 'View'", () => {
        const tree = renderer
            .create(
                <ButtonControl />,
            ).toJSON();
        expect(tree.type).toBe('View');
    });

    it('can hold a child', () => {
        const tree = renderer
            .create(
                <ButtonControl>
                    <Text>
                        Hello Test
                    </Text>
                </ButtonControl>,
            ).toJSON();
        expect(tree.children.length).toBe(1);
    });
});