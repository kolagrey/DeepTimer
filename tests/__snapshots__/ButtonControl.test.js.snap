// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonControl/> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "grey",
      "borderColor": "rgba(0, 0, 0, 0.29)",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
      "minWidth": 64,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "minHeight": 50,
            "minWidth": 100,
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "textTransform": "uppercase",
              },
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "fontFamily": "System",
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;
